pkg/server/auth/server.go

```diff
@@ -491,6 +491,16 @@ func contains(ss []string, s string) bool {
 	return false
 }
 
-func Error(rw http.ResponseWriter, err string, code int) {
-	http.Error(rw, fmt.Sprintf(`{"message": "%s"}`, err), code)
+func JSONError(w http.ResponseWriter, errStr string, code int) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(code)
+
+	response := struct {
+		Message string `json:"message"`
+	}{Message: errStr}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
```

Comment:
I think you want to just log the error if you fail to write it to the response.

Usual reason for failing to write it will be client-closed.